### PR TITLE
Feature/autocomplete export

### LINKED ISSUE #79: Feature Request: autocomplete for export in an imported module
Expected Behaviour:

```javascript
import {} from 'module';
```
when cursor is in  {} (or what ever valid format)
```javascript
import {_} from 'module';
```

It should show list of available exports
image

We can utilize this node library [get-exports-from-file](https://github.com/capaj/get-exports-from-file)

### PR DESCRIPTION
**Pull request by jonyeezs**
This resolves #79 

Will need to complete these tasks:

- [x] Re-organize code to fit export autocomplete
- [x] Autocomplete local `require`d exports
- [x] Autocomplete package `require`d exports
- [x] Autocomplete local `import`ed exports
- [x] Autocomplete package `import`ed exports

Will squash all commits when all tasks are completed.

### CODE REVIEW

**Comment by jonyeezs** [comment]
I reckon the code could use some refactoring after this gets merged.

I may revert some of the code in b64462c so it won't break any pending PRs.

File: src/utils/export-module-completion.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      results.map(entry => entry.name) :
+      lookupCommonJs(importModule, filePath))
+    .catch(() => {
+      return lookupCommonJs(importModule, filePath);
+    });
+};
+
+const lookupCommonJs = (importModule, projectPath) => {
+  const absoluteFile = path.resolve(projectPath, `${importModule}.js`);
+  return Object.keys(require(absoluteFile));
```
Comment (jonyeezs) [inline src/utils/export-module-completion.js]: I've come to a road-block.

Requiring a obsolute path in linux works like a charm.

But doesn't work on Windows. ARGH. Anyone knows why?

**Comment by jonyeezs** [comment]
Done! :fireworks: 

Unfortunately there aren't any unit test (none to begin with) so I had to do testing manually.

Works for most cases for me except a known issue when trying to autocomplete `{ | }` (where | is the cursor. See: https://github.com/atom/autocomplete-plus/issues/855

**Comment by jonyeezs** [comment]
> Works for most cases for me except a known issue when trying to autocomplete { | }

Further looking into it, maybe it's not a bug per se but a usage wise.

I'll have another commit to resolve that situation, that will be separate from the current.

**Comment by nkt** [comment]
Awesome, thanks!

**Pull request merged**